### PR TITLE
Fixed path based dev-dependencies break `install --no-dev` when the d…

### PR DESCRIPTION
**Pull request by jjuris**
Fixed path based dev-dependencies break `install --no-dev` when the directory path does not exist.

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

File: poetry/poetry.py [hunk 1]
```diff
@@ -147,7 +148,7 @@ def create(cls, cwd):  # type: (Path) -> Poetry
 
                 package.add_dependency(name, constraint)
 
-        if "dev-dependencies" in local_config:
+        if "dev-dependencies" in local_config and not "--no-dev" in sys.argv:
```
Comment (jjuris) [inline poetry/poetry.py]: Hm this change may affect all commands ( did not test it yet )

Comment (iScrE4m) [reply poetry/poetry.py]: Indeed, this doesn't look like the correct place to manage the behavior of the `install` command. What the `--no-dev` option does is it sets `Installer.dev_mode` attribute and you can probably work off of that.

Comment (jjuris) [reply poetry/poetry.py]: Hello,
yes you are right, this is a bad place, its more like patch / workaround. Unfortunately if you look closely into the code, you would find that class Poetry defined in this file is initialised in every single command, even the one which does not to anything effective, therefore it is required to change this control mechanism somehow
JJ.

**Comment by jjuris** [comment]
> You shouldn't modify version in `CHANGELOG.md`, `poetry/__version__.py`, `pyproject.toml`, leave that to the repo owner, who knows which version will be released.

Yes, its habit, this is not going to be merged, ignore it

**Comment by brycedrennan** [comment]
Closing since we were told to ignore.

**Pull request closed without merging**